cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -303,6 +328,174 @@ def migrate_from_modulestore(
             source_node=root_node,
         )
     change_log.save()
+    return change_log, root_migrated_node
+
+
+def _forwarding_content(migration: ModulestoreMigration, source_data: _MigrationSourceData) -> None:
+    """
+    Forwarding legacy content to migrated content
+    """
+    block_migrations = ModulestoreBlockMigration.objects.filter(overall_migration=migration)
+    block_sources_to_block_migrations = {
+        block_migration.source: block_migration for block_migration in block_migrations
+    }
+    for block_source, block_migration in block_sources_to_block_migrations.items():
+        block_source.forwarded = block_migration
+        block_source.save()
+
+    source_data.source.forwarded = migration
+    source_data.source.save()
+
+
+def _pupulate_collection(user_id: int, migration: ModulestoreMigration) -> None:
+    """
+    Assigning imported items to the specified collection in the migration
+    """
+    if migration.target_collection is None:
+        return
+
+    block_target_pks: list[int] = list(
+        ModulestoreBlockMigration.objects.filter(
+            overall_migration=migration
+        ).values_list('target_id', flat=True)
+    )
+    if block_target_pks:
+        authoring_api.add_to_collection(
+            learning_package_id=migration.target.pk,
+            key=migration.target_collection.key,
+            entities_qset=PublishableEntity.objects.filter(id__in=block_target_pks),
+            created_by=user_id,
+        )
+    else:
+        log.warning("No target entities found to add to collection")
+
+
+def _create_collection(library_key: LibraryLocatorV2, title: str) -> Collection:
+    """
+    Creates a collection in the given library
+    """
+    key = slugify(title)
+    collection = None
+    attempt = 0
+    while not collection:
+        modified_key = key if attempt == 0 else key + '-' + str(attempt)
+        try:
+            # Add transaction here to avoid TransactionManagementError on retry
+            with transaction.atomic():
+                collection = libraries_api.create_library_collection(
+                    library_key=library_key,
+                    collection_key=modified_key,
+                    title=title,
+                )
+        except libraries_api.LibraryCollectionAlreadyExists as e:
```

Comment:
Note: We are always creating new collections here. Therefore, if you migrate the same library multiple times, while the components are not duplicated (due to the default `skip` strategy), the collections are.
I'm not sure if there is an easy and pretty solution for this, though.

Reply:
Hi @rpenido, I created this https://github.com/openedx/edx-platform/pull/37381/commits/7503ce67378969aaba5f2a527772094d7f8105d0 with a partial solution (See the comment). Review it and tell me how you see it, before creating tests for that.